internal/security/secretstore/init.go

```diff
@@ -440,6 +464,61 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 
 }
 
+func getKnownSecretsToAdd() (map[string][]string, error) {
+	// Process the env var for adding known secrets to the specified services' secret stores.
+	// Format of the env var value is:
+	//   "<secretName>[<serviceName>;<serviceName>; ...], <secretName>[<serviceName>;<serviceName>; ...], ..."
+	knownSecretsToAdd := map[string][]string{}
+
+	addKnownSecretsValue := strings.TrimSpace(os.Getenv(addKnownSecretsEnv))
+	if len(addKnownSecretsValue) == 0 {
+		return knownSecretsToAdd, nil
+	}
+
+	serviceNameRegx := regexp.MustCompile(ServiceNameValidationRegx)
+	knownSecrets := strings.Split(addKnownSecretsValue, ",")
+	for _, secretSpec := range knownSecrets {
+		// each secretSpec has format of "<secretName>[<serviceName>;<serviceName>; ...]"
+		secretItems := strings.Split(secretSpec, "[")
+		if len(secretItems) != 2 {
+			return nil, fmt.Errorf("invalid specification for %s environment vaiable: Format of value '%s' is invalid. Missing or too many '['", addKnownSecretsEnv, secretSpec)
+		}
+
+		secretName := strings.TrimSpace(secretItems[0])
+
+		_, valid := validKnownSecrets[secretName]
+		if !valid {
+			return nil, fmt.Errorf("invalid specification for %s environment vaiable: '%s' is not a known secret", addKnownSecretsEnv, secretName)
+		}
+
+		serviceNameList := secretItems[1]
+		if !strings.Contains(serviceNameList, "]") {
+			return nil, fmt.Errorf("invalid specification for %s environment vaiable: Service list for '%s' missing closing ']'", addKnownSecretsEnv, secretName)
+		}
+
+		serviceNameList = strings.TrimSpace(strings.Replace(serviceNameList, "]", "", 1))
+		if len(serviceNameList) == 0 {
+			return nil, fmt.Errorf("invalid specification for %s environment vaiable: Service name list for '%s' is empty.", addKnownSecretsEnv, secretName)
+		}
+
+		serviceNames := strings.Split(serviceNameList, ";")
+		for index := range serviceNames {
+			serviceNames[index] = strings.TrimSpace(serviceNames[index])
+
+			if !serviceNameRegx.MatchString(serviceNames[index]) {
+				return nil, fmt.Errorf("invalid specification for %s environment vaiable: Service name '%s' has invalid characters.", addKnownSecretsEnv, serviceNames[index])
+			}
+		}
+
+		// This supports listing known secret multiple times.
+		// Same service name listed twice is not an issue since the add logic checks if the secret is already present.
+		existingServices := knownSecretsToAdd[secretName]
+		knownSecretsToAdd[secretName] = append(existingServices, serviceNames...)
+	}
+
+	return knownSecretsToAdd, nil
+}
```

Comment:
this logic seems too tedious to parse all of that. would a regular expression to parse out things better?

Reply:
If you can come up with a regular expression for this, I am willing to give it a try, but since it is multi level, I think that will be all most impossible

Reply:
I tried a couple online regex generator with no luck. Need to move on.

Reply:
@lenny-intel      tried to play in go playground here: https://play.golang.org/p/HIGjo_dnYRS   getting closer. you can try it out there.

Reply:
The multi layer dynamic lists make it too complex. The output of what you have has to be parsed and has lost the context. This function work and isn't too tedious. Let move on.

Reply:
I think it works if we can split by "," first then each of slice shares the same regex pattern that I just have in that go playground

Reply:
Yes, it works on good value, but no error feedback on bad values.

Reply:
Ok, i'll call it more lenient to error but i'm ok with your point on more robust handling here. ;-)